index.d.ts

```diff
@@ -0,0 +1,267 @@
+import EventEmitter from 'events';
+import { PassThrough, Writable } from 'stream';
+
+type EventsType = {
+  PASSED: 'passed';
+  FAILED: 'failed';
+  SKIPPED: 'skipped';
+  STARTED: 'started';
+  FINISHED: 'finished';
+}
+
+type zUnitOptions = Partial<{
+  defaults: unknown;
+  initial: unknown;
+  runtime: unknown;
+  bequeathed: unknown;
+}>;
```

Comment:
zUnit Options is a container for **scoped** Harness/Suite/Test/Hook options. The scoping is important for working out precedence if the same option is supplied to both a Suite and Test. For example
```js
const suite = new Suite('My Suite', { timeout: 1000 });
const test = new Test('My Test', { timeout: 3000 });
const suite.add(test);
const harness = new Harness(suite).run({ timeout: 500 });
```

In the above scenario the harness has been configured with runtime options, which should take precedence over the Suite and Test initialisation (initial) options.

The supported options are...

1. timeout (milliseconds) - Harness, Suite, Test & Hook
2. exclusive (boolean) - Suite & Test
3. skip (boolean) - Suite & Test
4. reason (string) - Suite & Test
5. pattern (regex) - Suite
6. directory (string) - Suite

Reply:
Sorted, to my knowledge...